samcli/local/apigw/authorizers/lambda_authorizer.py

```diff
@@ -284,3 +297,235 @@ def _parse_identity_sources(self, identity_sources: List[str]) -> None:
                     self._identity_sources.append(identity_source_validator)
 
                     break
+
+    def is_valid_response(self, response: str, method_arn: str) -> bool:
+        """
+        Validates whether a Lambda authorizer request is authenticated or not.
+
+        Parameters
+        ----------
+        response: str
+            JSON string containing the output from a Lambda authorizer
+        method_arn: str
+            The method ARN of the route that invoked the Lambda authorizer
+
+        Returns
+        -------
+        bool
+            True if the request is properly authenticated
+        """
+        try:
+            json_response = loads(response)
+        except (ValueError, JSONDecodeError):
+            raise InvalidLambdaAuthorizerResponse(
+                f"Authorizer {self.authorizer_name} return an invalid response payload"
+            )
+
+        if self.payload_version == LambdaAuthorizer.PAYLOAD_V2 and self.use_simple_response:
+            return self._validate_simple_response(json_response)
+
+        return self._validate_iam_response(json_response, method_arn)
+
+    def _validate_iam_response(self, response: dict, method_arn: str) -> bool:
+        """
+        Helper method to validate if a Lambda authorizer response is valid and authorized
+
+        Parameters
+        ----------
+        response: dict
+            JSON object from Lambda authorizer output
+        method_arn: str
+            The method ARN of this request
+
+        Returns
+        -------
+        bool
+            True if the request is authorized
+        """
+
+        def is_resource_authorized(response: dict, method_arn: str) -> bool:
```

Comment:
Question: why define this as a nested function instead of just another class method?

Reply:
Initial choice to leave this as an inner function was that since it was only being used here, moving it to a static class method didn't make that much sense to me. I can move this out though to make the tests easier to read/write.

Reply:
Oh this class is instantiated, scratch what I said about static class method 😆